aten/src/ATen/templates/ATenDispatch.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <c10/core/Backend.h>
+#include <unordered_map>
+
+// This dispatch class serves as a replacement for Type based dispatch.
```

Comment:
Try not to write comments in reference to historically previous concepts which no longer exist in the codebase. New readers aren't going to know what "Type based dispatch" means! Explain it from first principles.